runtime/src/main/java/org/corfudb/runtime/clients/LogUnitHandler.java

```diff
@@ -0,0 +1,207 @@
+package org.corfudb.runtime.clients;
+
+import io.netty.channel.ChannelHandlerContext;
+
+import java.lang.invoke.MethodHandles;
+
+import lombok.Getter;
+import lombok.Setter;
+import org.corfudb.protocols.wireprotocol.CorfuMsg;
+import org.corfudb.protocols.wireprotocol.CorfuMsgType;
+import org.corfudb.protocols.wireprotocol.CorfuPayloadMsg;
+import org.corfudb.protocols.wireprotocol.ReadResponse;
+import org.corfudb.runtime.exceptions.DataCorruptionException;
+import org.corfudb.runtime.exceptions.DataOutrankedException;
+import org.corfudb.runtime.exceptions.OutOfSpaceException;
+import org.corfudb.runtime.exceptions.OverwriteException;
+import org.corfudb.runtime.exceptions.TrimmedException;
+import org.corfudb.runtime.exceptions.ValueAdoptedException;
+
+
+/**
+ * A client to a LogUnit.
+ *
+ * <p>This class provides access to operations on a remote log unit.
+ * <p>Created by zlokhandwala on 2/20/18.
+ */
+public class LogUnitHandler implements IClient, IHandler<LogUnitClient> {
+
+    @Setter
+    @Getter
+    IClientRouter router;
+
+    @Override
+    public LogUnitClient getClient(long epoch) {
+        return new LogUnitClient(router, epoch);
+    }
+
+    /**
+     * The handler and handlers which implement this client.
+     */
+    @Getter
+    public ClientMsgHandler msgHandler = new ClientMsgHandler(this)
+            .generateHandlers(MethodHandles.lookup(), this);
+
+    /**
+     * Handle an WRITE_OK message.
+     *
+     * @param msg Incoming Message
+     * @param ctx Context
+     * @param r   Router
+     * @return True, since this indicates success.
+     */
+    @ClientHandler(type = CorfuMsgType.WRITE_OK)
+    private static Object handleOk(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r) {
+        return true;
+    }
+
+    /**
+     * Handle an ERROR_TRIMMED message.
+     *
+     * @param msg Incoming Message
+     * @param ctx Context
+     * @param r   Router
+     * @throws Exception Throws TrimmedException if address has already been trimmed.
+     */
+    @ClientHandler(type = CorfuMsgType.ERROR_TRIMMED)
+    private static Object handleTrimmed(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r)
+            throws Exception {
+        throw new TrimmedException();
+    }
+
+    /**
+     * Handle an ERROR_OVERWRITE message.
+     *
+     * @param msg Incoming Message
+     * @param ctx Context
+     * @param r   Router
+     * @throws OverwriteException Throws OverwriteException if address has already been written to.
+     */
+    @ClientHandler(type = CorfuMsgType.ERROR_OVERWRITE)
+    private static Object handleOverwrite(CorfuMsg msg, ChannelHandlerContext ctx, IClientRouter r)
+            throws Exception {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)